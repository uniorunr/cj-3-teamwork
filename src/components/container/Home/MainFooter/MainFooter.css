.main-footer {
  flex-direction: column;
  height: 25%;
}

.main-footer h2 {
  font-size: 18px;
  margin: 15px 0 25px;
}

.main-footer .developer-container {
  width: 70%;
  display: flex;
  justify-content: space-around;
  align-items: center;
}
