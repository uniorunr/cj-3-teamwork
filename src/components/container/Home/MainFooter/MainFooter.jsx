import React from 'react';
import PropTypes from 'prop-types';
import uniqid from 'uniqid';
import DeveloperItem from '../../../presentetion/DeveloperItem/DeveloperItem';
import './MainFooter.css';

const MainFooter = ({ teams, lang }) => (
  <div className="main-footer">
    <h2>{lang.footer ? lang.footer.title : 'not translate'}</h2>
    <div className="developer-container">
      {teams.map(user => (
        <DeveloperItem
          key={uniqid()}
          img={
            user.img
              ? user.img
              : 'https://cdn0.iconfinder.com/data/icons/Hand_Drawn_Web_Icon_Set/128/user.png'
          }
          name={user.name}
        />
      ))}
    </div>
  </div>
);

MainFooter.propTypes = {
  teams: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string,
      img: PropTypes.string,
    }),
  ),
  lang: PropTypes.instanceOf(Object).isRequired,
};

MainFooter.defaultProps = {
  teams: 'teams',
};

export default MainFooter;
