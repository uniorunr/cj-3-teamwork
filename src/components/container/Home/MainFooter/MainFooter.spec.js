import React from 'react';
import renderer from 'react-test-renderer';
import MainFooter from './MainFooter';

it('renders correctly', () => {
  const tree = renderer
    .create(<MainFooter teams={[{ name: 'a', img: 'img1' }, { name: 'b', img: 'img2' }]} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
