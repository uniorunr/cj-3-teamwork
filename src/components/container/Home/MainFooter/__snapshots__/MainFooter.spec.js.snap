// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="main-footer"
>
  <h2>
    Команда разработчиков
  </h2>
  <div
    className="developer-container"
  >
    <div
      className="developer-item"
    >
      <img
        alt="a"
        src="img1"
      />
      <p>
        a
      </p>
    </div>
    <div
      className="developer-item"
    >
      <img
        alt="b"
        src="img2"
      />
      <p>
        b
      </p>
    </div>
  </div>
</div>
`;
